### PR TITLE
expose alias names which can be 'require'd

### PR DESCRIPTION
**Pull request by peter-mouland**
Added global Alias for browserified JS files.
Added vendorBundle option for external files

### CODE REVIEW

**Comment by peter-mouland** [comment]
add `vendorBundle` into your `browserify` config object within `component.config.js`
```javascript
...
  browserify: {
        vendorBundle: [
            { file: './bower_components/d3/d3.js', expose: 'd3'}
        ]
    },
...
```

Match this wil the `browser` object within your `package.json`
```javascript
...
  "browser": {
    "d3": "./bower_components/d3/d3.js"
  },
...
```

after the `component build` command has finished you will have a `vendor.js` and `vendor.min.js` which contains all the files in the vendorBundle array. Within your source code you can require them individually `require('d3')`.

**Comment by peter-mouland** [comment]
crap - sorry. tried to be helpful and get it up to date with master (hoping a single `merge with master` commit message would appear).


The upside is that you can get across all the commits, but let me know if you need it rolled back to the actual PR again!

File: API.md [hunk 1]
```diff
@@ -59,12 +59,23 @@ The build can be configured more by adding an appropriately named object to the
     browserify: {
         insertGloabals : false,
         detectGlobals : false,
-        external: [
-            './bower_components/jquery/dist/jquery.js',
+        vendorBundle: [
+            { file: './bower_components/d3/d3.js', expose: 'd3'}
```
Comment (behaghel) [inline API.md]: Just to be sure I understand: the `expose` attribute is what I'll need to `require` this component in my code, right?

File: API.md [hunk 2]
```diff
@@ -59,12 +59,23 @@ The build can be configured more by adding an appropriately named object to the
...
     },
 ```
 
+The `vendorBundle` option will create `vendor.js` from external files. *(`debowerify` will sometime cause problems).*
+Ensure you have the corresponding `browser` object in your `package.json`. i.e.
+
+```javascript
+...
+  "browser": {
+    "d3": "./bower_components/d3/d3.js"
```
Comment (behaghel) [inline API.md]: I don't understand why it's needed. Particularly do you need to have the key `d3` to be the same as in the `vendorBundle` object or is it just a good practice for it's consistent?

File: component-structure/component.config.js [hunk 3]
```diff
@@ -13,8 +13,8 @@ module.exports = {
     test: 'karma', // or false. mocha not yet available.
     release: 's3', // or false.
     serve: 'staticApp', // `staticApp` or `nodeApp`
-    browserify: { //optional config
-        insertGloabals : false,
+    browserify: {
+        insertGlobals : true,
         detectGlobals : false
```
Comment (behaghel) [inline component-structure/component.config.js]: oh btw, I lied that day both used to be `true` by default. I probably screwed up my test without `detectGlobals: false` on sky-tags that day, in truth it doesn't work without it being `true`.

File: tasks/wrappers/browserify.js [hunk 4]
```diff
@@ -12,15 +12,36 @@ function Browserify(location, destination, options){
     this.options = options;
 }
 
+Browserify.prototype.buildVendor = function(options){
```
Comment (behaghel) [inline tasks/wrappers/browserify.js]: Shouldn't we add few more test-cases to `browserify.spec.js`?
- should create a vendor.js
- should expose component
- ...

File: API.md (hunk 1 above)
Comment (behaghel) [inline API.md]: you clarified that in your comment above, ignore me here.

File: API.md (hunk 2 above)
Comment (peter-mouland) [inline API.md]: i didn't get behind why, jus that you need to ensure the objects tally up

**Pull request merged**